polaris-core/src/main/java/org/apache/polaris/core/storage/aws/AwsStorageConfigurationInfo.java

```diff
@@ -122,7 +126,8 @@ public String getAwsAccountId() {
     if (arn != null) {
       Matcher matcher = ROLE_ARN_PATTERN_COMPILED.matcher(arn);
       checkState(matcher.matches());
-      return matcher.group(2);
+      // group(3) is the account identifier (either 12-digit AWS account or vendor namespace)
```

Comment:
The `getAwsAccountId` method name becomes misleading now that the value may not be an AWS accound ID. Please rename.